firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/FieldMask.java

```diff
@@ -26,13 +26,13 @@
  * object foo. If foo is not an object, foo is replaced with an object containing foo.
  */
 public final class FieldMask {
-  public static FieldMask fromCollection(Collection<FieldPath> mask) {
+  public static FieldMask fromSet(Set<FieldPath> mask) {
     return new FieldMask(mask);
   }
 
-  private final Collection<FieldPath> mask;
+  private final Set<FieldPath> mask;
```

Comment:
Optional: what's the reason to prefer `Set` to a `List`? AFAICS, `mask` is never used for lookup; the only access is iterating over all the elements (line 63), for which `List` will likely be faster.

Reply:
I agree that Set is the natural data structure (and perf, etc. aren't major concerns).  That said, we may now get slightly different behavior than on other platforms (e.g. with regard to duplicates), though I don't think that's a big deal, and the Android behavior will probably be more correct.

Reply:
I'm fine with any of these options (including keeping it as a Set as you have it now).